Visualization/CombineBlastTables.R

```diff
@@ -0,0 +1,120 @@
+# Copyright Jackson M. Tsuji, 2018
+# Neufeld Research Group
+# See required R packages below.
+
+#####################################################
+## User variables: #################################
+RUN_COMMAND_LINE <- TRUE # If selected, all user input here is ignored, and terminal-based input is expected instead.
+
+# Set other user variables here
+if (RUN_COMMAND_LINE == FALSE) {
+  setwd("/Users/JTsuji/Downloads") # your working directory where files are stored
+  input_filenames <- c("subject1.csv", "subject2.csv", "subject3.csv") # NOTE: filename minux extesion will become the subject_name!
+  output_filename <- "combined_blast_tables.csv"
+}
+#####################################################
+
+#####################################################
+## Load required packages: ##########################
+library(getopt)
+library(plyr)
+suppressMessages(library(dplyr))
+#####################################################
+
+# Hard-coded variables
+header_names <- c("qseqid", "sseqid", "pident", "evalue", "qcovhsp", "bitscore")
+
+
+# Function to assign command line input to variables, or throw help message
+parse_command_line_input <- function() {
+  
+  inputArgs <- commandArgs(trailingOnly = TRUE)
+  if (length(inputArgs) < 2) {
+    cat("CombineBlastTables.R: Combines BLAST tables and makes new column for sample ID based on the filenames.\n")
+    cat("Copyright Neufeld Lab, 2018\n")
+    cat("Contact Jackson M. Tsuji (jackson.@uwaterloo.ca) for bug reports or feature requests.\n\n")
+    
+    cat("Usage: CombineBlastTables.R subject1.csv subject2.csv ... subjectN.csv combined_blast_tables.csv\n\n")
+    
+    cat("Details:\n",
+        "subject1.csv, etc.: CSV BLAST tables for all individual samples (subjects for BLAST). Name should be the subject name [Required]\n",
+        "combined_blast_tables.csv: the output CSV file. [Required]\n\n")
+        
+    quit(status = 1)
+  }
+  
+  num_args <- length(inputArgs)
+  
+  # Make variables from provided input and save as global variables (<<-)
+  input_filenames <<- inputArgs[1:(num_args-1)]
+  output_filename <<- inputArgs[num_args]
+  
+}
+
+# Helper function for timestamps
+ts <- function() {
+  
+  datetime_utc <- format(as.POSIXlt(Sys.time(), tz = "UTC"), "%a %d %b %Y %H:%M:%S %Z")
+  
+  date_message <- paste("[ ", datetime_utc, " ]: ", sep = "")
+  
+  return(date_message)
+}
+
+
+# Function to load table of data, add header IDs, and add query/subject name by parsing filename
+load_individual_table <- function(table_filename, header_names) {
+  cat(paste(ts(), "Reading table '", table_filename, "'\n", sep = ""))
+  
+  # Read the table and add columns
+  data_table <- read.table(table_filename, header = FALSE, sep = ",", stringsAsFactors = FALSE, comment.char = "")
+  colnames(data_table) <- header_names
+  
+  # Remove CSV ending and folder path. Set to subject_name.
+  table_filename_base <- gsub(pattern = ".csv$", replacement = "", x = table_filename)
```

Comment:
Replace these with R built-in pathing functions:

- https://stat.ethz.ch/R-manual/R-devel/library/base/html/basename.html
- https://stackoverflow.com/questions/14173754/splitting-a-file-name-into-name-extension

The reason you want to use the built-in path functions is that they will allow your scripts to work on other platforms (ex. Windows) which use `\` instead of `/` with no effort on your part. Also, it will take into account things like `~` and `$HOME` in your paths and expand them.